### DISCUSSION TITLE
How are the release assets generated ?

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [emgucv/emgucv#490] [Q&A]
What are the commands that generate the three `libemgucv-` assets in the GitHub release ?

**Comment by aleks-ivanov** [comment]
@emgucv ping

We are working on a CI/CD pipeline for EmguCV with automated versioning and NuGet/GitHub Release publishing and to be sure all the resources are properly generated to be used for the deployment tasks, we'd like to test everything before contributing 🙂 

Also while on the topic of the automated versioning, is there currently a variable in the build scripts, that could be used to set/overwrite the version during the processes that generate the NuGet packages and release assets ? e.g. `dotnet build asd.csproj -p:Version=$CustomVersion`

**Reply by emgucv** [reply]
The cmake build script detect the open cv version as well as git commit count and produce the version number:

{opencv_major}.{opencv_minor}.{opencv_patch}.{emgucv_git_commit_height}

It is also used to generate the version number in the "Package.nuspec" files used for building the nuget packages.

**Reply by aleks-ivanov** [reply]
Understood, so you already have automated versioning. Got it 🙂 

So, to the original question to which I should've pointed in my last comment, what exactly are the commands/scripts that generate all the packages that are published to NuGet and the three files that are attached as assets to the GitHub release ? 
Also are there any other specific tool prerequisites other than .NET SDK, CMAKE, NSIS and CUDA ?
